google/validation.go

```diff
@@ -0,0 +1,16 @@
+package google
+
+import (
+	"fmt"
+	"regexp"
+)
+
+func validateGCPName(v interface{}, k string) (ws []string, errors []error) {
```

Comment:
Good candidate for unit testing here

Also what do you think about splitting it out into two functions - `validateGCPName` and `validateRegexp` (where `validateGCPName` is implemented in terms of `validateRegexp`)? I can see us needing regexs for other things later (for example, metadata keys)